drake/examples/Cars/simulateLCM.cpp

```diff
@@ -78,11 +89,22 @@ int main(int argc, char* argv[]) {
   // be reused
   DrakeJoint::FloatingBaseType floating_base_type = DrakeJoint::QUATERNION;
 
-
   auto rigid_body_sys = make_shared<RigidBodySystem>();
-  rigid_body_sys->addRobotFromFile(argv[1], floating_base_type);
-  auto const & tree = rigid_body_sys->getRigidBodyTree();
-  for (int i=2; i<argc; i++)
+
+  // The following variable, weld_to_frame, is only needed if the model is a
+  // URDF file. It is needed since URDF does not specify the location and
+  // orientation of the car's root node in the world. If the model is an SDF,
+  // weld_to_frame is ignored by the parser.
```

Comment:
I agree that the `RigidBodySystem::addRobotFromFile()` API is confusing. In the current implementation, the `weld_to_frame` parameter is ignored if the model is an SDF, so it would not be possible for there to be an additional offset of the root link of the robot. However, including parameters that are ignored under certain circumstances is definitely bad.

Another concern is the fact the URDF and SDF are very different in terms of the number of robots it can contain. URDF contains a single robot while SDF may contain numerous robots. Having a single `RigidBodySystem::addRobotFromFile()` method for handling both types of files no longer seems appropriate.

Given the above observations, here's how I propose to fix the problem. Let's remove the logic within `RigidBodySystem::addRobotFromFile()` that attempts to automatically determine whether a model file is a URDF or SDF and instead push that logic into the creator of `RigidBodySystem`, which in this case is `drake/examples/Cars/simulateLCP.cpp`. 

Thus, replace the current `RigidBodySystem::addRobotFromFile()` method:

```
void addRobotFromFile(
      const std::string& filename,
      const DrakeJoint::FloatingBaseType floating_base_type =
          DrakeJoint::QUATERNION,
      std::shared_ptr<RigidBodyFrame> weld_to_frame = nullptr);
```

with the following two methods:

```
void AddRobotFromURDFFile(
      const std::string& file_name,
      const DrakeJoint::FloatingBaseType floating_base_type =
          DrakeJoint::QUATERNION,
      const Eigen::Vector3d floating_base_rpy = Eigen::Vector3d::Zero(),
      const Eigen::Vector3d floating_base_xyz = Eigen::Vector3d::Zero());
```
and
```
void AddRobotFromSDFFile(
      const std::string& file_name, const std::string & robot_name
      const DrakeJoint::FloatingBaseType floating_base_type =
          DrakeJoint::QUATERNION);
```
Notice that `AddRobotFromURDFFile()` includes parameters for defining the initial state of the 6-DOF virtual floating joints, while `AddRobotFromSDFFile()` includes a parameter called `robot_name` that identifies which robot from within the SDF file to load. This design will require that `AddRobotFromSDFFile()` be called multiple times, one for each robot in the SDF file. I think this is OK since it lends more control to the user and allows certain models inside the SDF file to be treated differently, e.g., as a robot vs. a height map. If pessary, we can always add a convenience method called `AddAllRobotsFromSDFFile()` that automatically loads all models in an SDF file.

What do you think about the above re-design?